x/request/user.go

```diff
@@ -0,0 +1,47 @@
+package request
+
+import "context"
+
+const authenticatedUserKey = contextValueKey("authenticatedUser")
+
+// AuthenticatedUser holds the identifiers of a user making an authenticated
+// request.
+type AuthenticatedUser struct {
+	AccountAggregateID       string
+	RealUserAggregateID      string
+	EffectiveUserAggregateID string
```

Comment:
```suggestion
	CustomerAccountID       string
	UserID string
	RealUserID      string
```

I don't think adding `Aggregate` in here is helpful. For JWT auth, that's what they're going to be set to, but I'm not convinced it should be part of the field name.

Reply:
I'm happy with the shorter names; I wasn't sure how useful being explicit with `aggregate` would be.

Reply:
Would it make sense to use the same names as web gateway? https://github.com/cultureamp/web-gateway/blob/ccf9db2fabb2a798a9905475a3fce63fdc192811/identity/identity.go#L20-L24

Reply:
That is what's exposed in the JWT. I'd follow that _except_ for `EffectiveUserID` becomes `UserID`. Every single time I see `Effective` and `Real` I have to ask myself which one I'm going to use. Even `Real` is ambiguous.

`UserID` and `ImpersonatingUserID` might be better.

Reply:
Yeah, `Real` is only real in context. Would `ImpersonatingUserID` be the ID of the person who is impersonating, or being impersonated?

Naming is hard.

Reply:
I'm also happy to keep it as `AccountID`, `UserID`, and `RealUserID` to be consistent with other systems for now.